assets/src/helpers.js

```diff
@@ -434,41 +433,62 @@ export const getPercentageFromPixels = ( axis, pixelValue ) => {
-	const currentPost = select( 'core/editor' ).getCurrentPost();
-	const editedFeaturedMedia = select( 'core/editor' ).getEditedPostAttribute( 'featured_media' );
-	const featuredMedia = currentPost.featured_media || editedFeaturedMedia;
-
-	if ( ! featuredMedia ) {
-		return __( 'Selecting a featured image is required.', 'amp' );
-	}
-
-	const media = select( 'core' ).getMedia( featuredMedia );
-	if ( ! media || ! media.media_details || ! validateImageSize( media.media_details ) ) {
-		return invalidSizeMessage;
-	}
+export const getMinimumFeaturedImageDimensions = () => {
+	// "Images should be at least 1200 pixels wide."
+	const width = 1200;
+
+	// "For best results, provide multiple high-resolution images (minimum of 800,000 pixels
+	// when multiplying width and height) with the following aspect ratios: 16x9, 4x3, and 1x1."
+	// Given this requirement, make ensure that the image can successfully be cropped into such
+	// an aspect ratio by making sure that it has the. The 16/9 aspect ratio is chosen because it
+	// has the smallest height for the given width.
+	const height = width * ( 9 / 16 );
+
+	return { width, height };
 };
 
 /**
- * Gets whether the AMP story's featured image has the right minimum dimensions.
+ * Determines whether whether the image has the minimum width for an AMP story featured image.
  *
  * The featured image will be used for the poster-portrait-src.
- * It should have a width of at least 696px and a height of at least 928px.
+ * For this, it should have a width of at least 696px and a height of at least 928px.
+ *
+ * It also should have a width of at least 1200 pixels
+ * to satisfy the requirement of Google Search for Schema.org metadata.
  *
  * @param {Object} media A media object with width and height values.
  * @return {boolean} Whether the media has the minimum dimensions.
  */
 export const hasMinimumStoryPosterDimensions = ( media ) => {
-	const minWidth = 696;
-	const minHeight = 928;
+	const minWidth = 696; // @todo Needs to incorporate Math.max( 696, getMinimumFeaturedImageDimensions().width );
+	const minHeight = 928; // @todo This needs to be Math.max( 928, getMinimumFeaturedImageDimensions().height ).
+
 	return (
 		( media.width && media.height )	&&
 		( media.width >= minWidth && media.height >= minHeight )
 	);
 };
+
+/**
+ * Whether the image meets the requirements for use as featured image.
+ *
+ * This should satisfy the requirements of Google Search for Schema.org metadata.
+ *
+ *
+ * @param {Object} media A media object with width and height values.
+ * @return {boolean} Whether the media has the minimum dimensions.
+ */
+export const hasMinimumFeaturedImageWidth = ( media ) => {
+	if ( ! media || ! media.width || ! media.height ) {
+		return false;
+	}
+	const minFeaturedImageDimensions = getMinimumFeaturedImageDimensions();
+
+	return media.width >= minFeaturedImageDimensions.width && media.height >= minFeaturedImageDimensions.height;
```

Comment:
Instead of returning a `bool` I wonder if it would be better to return a string for the error message. An empty string or null could mean there is no error. By returning the error message, it could then indicate why specifically the image is not adequate, and it can indicate what the dimensions are for the `media` supplied.

Reply:
Besides the message itself it would be nice to return the affected data (e.g. actual width vs required width) as well, so it could be more like returning/throwing an error object.

Will there be more error messages than "image too small" and "no image set" in the future? Perhaps "wrong file format" and "file too big"? This could also affect changes to this function here.

Reply:
Yes, exactly. The other constraint not currently being accounted for is that the image must be a JPEG, PNG, or GIF. So that is something this function should check for. In the future the requirements will surely change, so if we keep it flexible then it adapt. In that regard, how about having this return an _array_ of errors? Then we can return an error regarding the file type, insufficient width, and insufficient height. The user then can fix all at once and not have to discover after fixing one error that there is yet another error to fix.